### PR TITLE
[SMS delivery pickup] Build basic UI for accepting delivery

### LINKED ISSUE #122: [SMS delivery pickup] Build basic UI for accepting delivery
## Summary
Build out UI for accepting deliveries from the [delivery needed map. ](https://crownheightsma.herokuapp.com/delivery-needed)

## Context
Design doc: https://docs.google.com/document/d/1wSWJHs_MR_4JUcKIwlp3pt_kWB0JuG0ixMXLClZ2qIM/edit

## Details
- Add `Claim delivery` button on popover. e.g.
Screen Shot 2020-08-01 at 12 09 22 PM

- On `Claim delivery` button click, volunteer should be brought to a Responsibilities view detailing what delivery means.
  - This may be a modal, or a new view, depending on content.
    - Put placeholder while we work out what intake wants to put there

- On the Responsibilities view, there should be a field for a phone number and a`Yes, I accept` or a `No, I need more info` button
   - `Yes` button should only be clickable once phone number has been entered.
     -  This can become repetitive quickly to enter a phone number again and again. Consider way to store locally but securely. Maybe for enhancement, not necessary for this PR.
  - `No` button should link out to a resource or way to contact #wg-intake

- Once `Yes` button has been clicked, fire an async request to endpoint (TBA)
  - On success, display a thank you modal detailing next steps
     - Remove the current delivery from map (a refresh might be sufficient here)
  - On failure, direct volunteer to #tech-support on Slack

Screen Shot 2020-08-01 at 12 12 38 PM

Notes:
- Build out as much of this as possible using Material UI, which should have modals, form elemetns and whatnot for this view
- My sketches for the UI are very rough, please refine as you see fit!

### LINKED ISSUE #133: [SMS Pickup] Hide Claim Delivery button behind a URL param
Blocked by https://github.com/crownheightsaid/mutual-aid-app/issues/122

## Why
We want to release this feature to seasoned delivery volunteers before it goes out to everyone, so we can iron out bugs if there are any.

## What
- `/delivery-needed` shouldn't show the button for `Claim Delivery` (https://github.com/crownheightsaid/mutual-aid-app/issues/122) by default. 
- button should only be rendered when `/delivery-needed?sms_pickup=true` URL param is present. 

## Considerations
- we'll likely make this button appear by default in the future once bugs are ironed out
- please note that other URL params e.g. `request=fsfds` might exist
- there is a[ helper function ](https://github.com/crownheightsaid/mutual-aid-app/blob/master/src/webapp/helpers/getRequestParam.js)to fetch request params in the codebase already

### PR DESCRIPTION
**Pull request by ifearcompilererrors**
TODO:
- [ ] ask intake for dialog copy
- [x] phone number field
- [x] phone number validation
- [x] "Yes" state
- [x] finish state
- [x] "No I need more info" state
- [x] hide claim delivery button when `sms_pickup=true` param is present

closes #122
closes #133

### CODE REVIEW

**Comment by piratefsh** [comment]
hey! i just took a quick look and everything looks good so far! i see that you introduced a context and there's some weird circular dependency happening. i'll also be introducing another context in a future PR that's for a different flow. i'm thinking we can just put contexts in `webapp/components/contexts`, that way it can be imported without circular dependency + reduce merge conflicts in the future.

**Comment by piratefsh** [comment]
i'll update my context to be in webapp/context so we don't end up with duplicated dirs!

**Comment by ifearcompilererrors** [comment]
@piratefsh woohoo! thanks for taking a look. Excited for more context wowow. The dialog UI probably looks a little rough without copy, but hoping it's a step in the right direction!

File: src/webapp/components/DeliveryDialog/InstructionsScreen.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+);
+
+const componentTypeMap = {
+  subtitle1: Subtitle,
+  p: Paragraph,
+  ul: UnorderedList,
+  li: ListItem,
+};
+
+const Instructions = () => {
```
Comment (ifearcompilererrors) [inline src/webapp/components/DeliveryDialog/InstructionsScreen.js:65]: lmk if i over-engineered this 👀

Comment (piratefsh) [reply src/webapp/components/DeliveryDialog/InstructionsScreen.js:65]: not at all! having instructions in a separate JSON file will make it easier for future copy updates. i only worry about doing this alongside the use of i18n might be confusing to other mutual aids that forked from us, so we should make sure to document why we're doing this (i imagine to allow for flexibility in structuring the content).

File: src/webapp/components/DeliveryDialog/InstructionsScreen.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { makeStyles } from "@material-ui/core/styles";
+import Typography from "@material-ui/core/Typography";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogActions from "@material-ui/core/DialogActions";
+import Button from "@material-ui/core/Button";
+import instrutions from "./instructions.json";
```
Comment (piratefsh) [inline src/webapp/components/DeliveryDialog/InstructionsScreen.js:9]: small nit: typo here

File: src/webapp/components/DeliveryDialog/InstructionsScreen.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          }),
+        };
+
+        return React.createElement(ComponentBlock, { ...opts });
+      })}
+    </>
+  );
+};
+
+const InstructionsStep = ({ handleGoBack }) => {
```
Comment (ifearcompilererrors) [inline src/webapp/components/DeliveryDialog/InstructionsScreen.js:85]: probably going to refactor InstructionsStep back into DialogScreens and keep the Instruction formatting logic in here

**Pull request merged**